### PR TITLE
docs(gh-pages): switch Travis to GH Actions

### PR DESCRIPTION
**Pull request by curbengh**
Free for public and private repository.
Supersedes https://github.com/hexojs/site/pull/1343

## Check List

**Please read and check followings before submitting a PR.**

- [ ] I want to publish my theme on Hexo official website.
    - [ ] I have read the [theme publishing doc](https://hexo.io/docs/themes#Publishing).
    - [ ] `name` is unique.
    - [ ] `link` URL is correct.
    - [ ] `preview` URL is correct.
    - [ ] `preview` URL web site is rendered correctly.
    - [ ] Add a screenshot to `source/themes/screenshots`.
    - [ ] Screenshot filename is same as value of `name`.
    - [ ] Screenshot size is `800 * 500`.
    - [ ] Screenshot file format is `png`.
- [ ] I want to publish my plugin on Hexo official website.
    - [ ] I have read the [plugin publishing doc](https://hexo.io/docs/plugins#Publishing).
    - [ ] `name` is unique.
    - [ ] `link` URL is correct.
- [x] Others (Update, fix, translation, etc...)
    - Languages:
    - [x] `en` English
    - [ ] `ko` Korean
    - [ ] `pt-br` Brazilian Portuguese
    - [ ] `ru` Russian
    - [ ] `th` Thai
    - [ ] `zh-cn` simplified Chinese
    - [ ] `zh-tw` traditional Chinese

### CODE REVIEW

File: source/docs/github-pages.md [hunk 1]
```diff
@@ -44,9 +60,39 @@ If you prefer to have a project page on GitHub:
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./public
+-          publish_branch: master  # deploying branch
++          publish_branch: gh-pages
+```
```
Comment (curbengh) [inline source/docs/github-pages.md:85]: WFM https://github.com/curbengh/hexo-testing

File: source/docs/github-pages.md [hunk 2]
```diff
@@ -2,39 +2,55 @@
...
+  ```
+3. Add `.github/workflows/pages.yml` file to your repo with the following content:
+
+```yml .github/workflows/pages.yml
+name: Pages
+
+on:
+  push:
+    branches:
+      - source  # default branch
```
Comment (curbengh) [inline source/docs/github-pages.md:21]: I haven't test this, seems strange to me. https://github.com/marketplace/actions/github-pages-action#%EF%B8%8F-repository-type---user-and-organization

Comment (SukkaW) [reply source/docs/github-pages.md:21]: FYI, I am using `peaceiris/actions-gh-pages@v2` which WFM.

Comment (curbengh) [reply source/docs/github-pages.md:21]: Are you using master+gh-pages (most common?) or source+master?

Comment (SukkaW) [reply source/docs/github-pages.md:21]: @curbengh 

`master` & `gh-pages`. While `peaceiris/actions-gh-pages@v2` supports both of them.

Also, GitHub is testing publishing github pages from any branch, so it won't matter in the future.

Comment (curbengh) [reply source/docs/github-pages.md:21]: > While `peaceiris/actions-gh-pages@v2` supports both of them.

I find it puzzling the [docs](https://github.com/marketplace/actions/github-pages-action#%EF%B8%8F-repository-type---user-and-organization) suggest switching to `source+master`, while `master+gh-pages` still works. Should I retain `source+master` in this PR?

Comment (SukkaW) [reply source/docs/github-pages.md:21]: > Should I retain `source+master` in this PR?

It would be fine.

**Comment by curbengh** [comment]
@SukkaW 
just noticed we don't have a style for `marked` class.

File: source/docs/github-pages.md (hunk 2 above)
Comment (curbengh) [reply source/docs/github-pages.md:38]: Looks like `User page` repo requires `source+master`. https://github.com/hexojs/site/issues/1324

**Pull request merged**